### DISCUSSION TITLE
How to use checkbox?

### DISCUSSION THREAD

**Discussion by minthemiddle** [filamentphp/filament#262] [Help]
How to structure the underlying data to use the checkbox in Filament?
I thought it would work with a boolean data structure but that does not work.

Migration:
```php
Schema::create('posts', function (Blueprint $table) {
…
$table->boolean('published');
});
```

PostResource:
```php
return $form
            ->schema([
                Checkbox::make('published'),
            ]);
    }
```

This will show the checkbox but on save it does not update the value. I put the `published` field in `$fillable`. I can update the boolean with the `select` component:

```php
return $form
            ->schema([
                Select::make('published')->options([
                   true => 'Published',
                   false => 'Private',
                ]),
            ]);
    }
```

**Comment by danharrin** [comment]
Hm what you tried should work, please open an issue with a demo repository if possible.

**Reply by minthemiddle** [reply]
Could reproduce with a new repo. Please see: https://github.com/laravel-filament/filament/issues/271
